### ISSUE TITLE
is-subset always returns true for date values

### ISSUE THREAD

**Issue by petergphillips** [studio-b12/is-subset#3]
For values that are dates is-subset will ignore the values and return true when the dates are different.

For example:

``` javascript
is-subset(
    { date, new Date(0)}
    { date, new Date()}
)
```

will return true even though the dates are different.  Looking at the code this is because Object.keys() for a Date will return an empty array.

Something like

``` javascript
  if (
    superset instanceof Date && subset instanceof Date
  ) return superset.valueOf() === subset.valueOf();
```

after the typeof object check should fix it.  I started trying to fix the issue but couldn't get any of the tests to run - getting TypeError: undefined is not a function for each test.

**Comment by tomek-he-him** [comment]
Hi, thanks for the heads up!

I’ll have a look at it next week when I’m back from vacation.

**Comment by tomek-he-him** [comment]
Closed through #6. (Thanks, @petergphillips!)
